app/uk/gov/hmrc/trackingconsentfrontend/config/ErrorHandler.scala

```diff
@@ -16,21 +16,23 @@
 
 package uk.gov.hmrc.trackingconsentfrontend.config
 
-import javax.inject.{Inject, Singleton}
+import javax.inject.Inject
 import play.api.i18n.MessagesApi
-import play.api.mvc.Request
+import play.api.mvc.RequestHeader
 import play.twirl.api.Html
 import uk.gov.hmrc.play.bootstrap.frontend.http.FrontendErrorHandler
 import uk.gov.hmrc.trackingconsentfrontend.views.html.components.ErrorTemplate
 
-@Singleton
-class ErrorHandler @Inject() (
-  val messagesApi: MessagesApi,
-  implicit val appConfig: AppConfig,
-  errorTemplate: ErrorTemplate
+import scala.concurrent.{ExecutionContext, Future}
+
+class ErrorHandler @Inject() (val messagesApi: MessagesApi, errorPage: ErrorTemplate)(using
+  appConfig: AppConfig,
+  protected val ec: ExecutionContext
 ) extends FrontendErrorHandler {
-  override def standardErrorTemplate(pageTitle: String, heading: String, message: String)(implicit
-    request: Request[_]
-  ): Html =
-    errorTemplate(pageTitle, heading, message)
+
+  override def standardErrorTemplate(pageTitle: String, heading: String, message: String)(using
+    request: RequestHeader
+  ): Future[Html] = Future {
+    errorPage(pageTitle, heading, message)
```

Comment:
I think we should use `Future.successful` here instead because we don't actually need to do the rendering of the error page async